### DISCUSSION TITLE
Hiding long sections partially

### DISCUSSION THREAD

**Discussion by ojacques** [squidfunk/mkdocs-material#5976] [Ideas]
Hello mkdocs-material community!

I write documentation and tutorials which may have long sections, between two options. For example:

```markdown
You could do it this way:

- long set of instructions

Or this other way:

- Another long set of instructions
```

I would like the ability to partially hide  long sections of texts.
I say "partially", as I still want to see an abstract of the text in this partially hidden section, so that my readers can judge whether they need to see and read what's next.

What is unique here is that the section is not entirely hidden, but reveals few lines, with a nice gradient.

An example on how this is done on [this AWS documentation site](https://docs.aws.amazon.com/systems-manager/latest/userguide/setup-instance-permissions.html):

![image](https://github.com/squidfunk/mkdocs-material/assets/5385290/114e6484-f272-4679-845e-e7ad66f74c26)

As you can see, some sections are partially hidden, until the reader presses "Show more", which unfolds the section.

![image](https://github.com/squidfunk/mkdocs-material/assets/5385290/1bf2a5ef-7ff0-4175-bbe7-30069bcc5a5b)

One can then click "Show less" to fold the section back.

I could not find a way to do this. It may be fiddling with [PyMDown details extension](https://facelessuser.github.io/pymdown-extensions/extensions/details) or [blocks](https://facelessuser.github.io/pymdown-extensions/extensions/blocks/), but I could not figured it out.

Any idea? Thanks.

**Comment by squidfunk** [comment]
You can open a feature request if you like – I may have an idea how this can be done in a good and compatible way ☺️
